### PR TITLE
feat: deserialize requests

### PR DESCRIPTION
**Pull request by prestwich**
## Motivation

Deserialize request objects

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

**Comment by mattsse** [comment]
@prestwich clippy does not like this -.- perhaps this works `MaybeUninit::<T>::zeroed().assume_init()`

**Review by DaniPopes — APPROVED** [review]

**Comment by DaniPopes** [comment]
It's not UB if it's zero-sized, fixable by just allowing the lint and removing the transmute since that lint is valid

File: crates/json-rpc/src/request.rs [hunk 1]
```diff
@@ -182,6 +186,107 @@ where
...
+                }
+
+                if params.is_none() {
+                    if std::mem::size_of::<Params>() == 0 {
+                        // SAFETY: params is a ZST, so it's safe to fail to initialize it
+                        unsafe {
+                            params = Some(std::mem::transmute(
+                                MaybeUninit::<Params>::uninit().assume_init(),
+                            ))
+                        }
```
[the remark this answers is not in the document]
Comment (prestwich) [reply crates/json-rpc/src/request.rs]: changed it to this and removed unnecessary transmute

**Comment by prestwich** [comment]
`zeroed` and `uninit` should be equivalent for a ZST, but zeroed won't trigger the lint, so I removed the transmute and changed to zeroed

**Pull request merged**